LaLauncher/src/main/java/com/github/postapczuk/lalauncher/MainActivity.java

```diff
@@ -52,7 +53,14 @@ public void fetchAppList() {
             String labelName = getApplicationLabel(componentName);
             if (labelName != null) apps.add(labelName);
         }
-        apps.add(ADD_APPLICATION);
+
+        // Show "+ Add favorite app" on initial run
+        // Shorten to "+" when any favorite is set
+        if(apps.size() > 0) {
```

Comment:
> Please do not add comments to the code. Code should be self descriptive

This is something some people say, but it doesn’t seem possible. It’s really night and day to get into a project with and without code comments.

There’s no simple way to see that the design spec is "Shorten to "+" when any favorite is set"

Reply:
I can't agree. If you extract the private method with proper naming it will be easy to understand. That is common approach and it's named "clean code"

Reply:
```java
private List<String> setAddFavouriteAppLabel(List<String> apps) {
    if(apps.size() > 0) {
        apps.add(ADD_APPLICATION_SHORT)
    } else {
        apps.add(ADD_APPLICATION);
    }
    return apps;
}
```